HW2/P4/P4.txt

```diff
@@ -0,0 +1,32 @@
+Original times:
+Main: 4.49185609818 seconds for 10 filter passes.
+
+
+Multithreaded times:
+
+See performancebynumthreads.png for a graph of the data
+
+Threads: 1. Time: 4.78213000298 seconds for 10 filter passes.
+Threads: 1. Time: 4.091766119 seconds for 10 filter passes.
+Threads: 1. Time: 4.1126601696 seconds for 10 filter passes.
+Threads: 1. Time: 4.14261603355 seconds for 10 filter passes.
+Threads: 1. Time: 4.11232304573 seconds for 10 filter passes.
+
+Threads: 2. Time: 2.27520394325 seconds for 10 filter passes.
+Threads: 2. Time: 2.28757190704 seconds for 10 filter passes.
+Threads: 2. Time: 2.28277492523 seconds for 10 filter passes.
+Threads: 2. Time: 2.2849111557 seconds for 10 filter passes.
+Threads: 2. Time: 2.28204512596 seconds for 10 filter passes.
+
+Threads: 4. Time: 2.11169791222 seconds for 10 filter passes.
+Threads: 4. Time: 2.10361099243 seconds for 10 filter passes.
+Threads: 4. Time: 2.1059319973 seconds for 10 filter passes.
+Threads: 4. Time: 2.11611104012 seconds for 10 filter passes.
+Threads: 4. Time: 2.1030728817 seconds for 10 filter passes.
+
+
+This shows that increasing from one to two threads leads to a substantial increase in performance, around 2x, but the increase to four threads does not substantially increase performance from two threads. I suspect this may have to do with my having a Macbook Air. That being said, there are slighy performance gains with 4 threads.
+
```

Comment:
You should look at the number of cores you have in you computer.
It is not obvious that it should get better from 2 to 4 thread if you have only one core, since it just adds some overhead.